engine.py

```diff
@@ -112,42 +113,52 @@ def execute_command(self, cmd_key, args):
         if cb_var_args:
             have_expected_args = (len(args) >= len(cb_arg_list))
         else:
-            have_expected_args = (len(args) == len(cb_arg_list)) 
-        
+            have_expected_args = (len(args) == len(cb_arg_list))
+
         if not have_expected_args:
             expected_args = list(cb_arg_list)
             if cb_var_args:
                 expected_args.append("*%s" % cb_var_args)
             raise TankError("Cannot run command! Expected command arguments (%s)" % ", ".join(expected_args))
-        
+
         if not self._has_qt:
             # QT not available - just run the command straight
             return cb(*args)
         else:
             from tank.platform.qt import QtCore, QtGui
-            
-            # we got QT capabilities. Start a QT app and fire the command into the app
-            tk_shell = self.import_module("tk_shell")
-            t = tk_shell.Task(self, cb, args)
-            
-            # start up our QApp now, if none is already running
-            qt_application = None
+
+            # start up our QApp now
```

Comment:
I think the creation of the `QApplication` and setting the icon was meant to go inside the `if`, any reason why you moved it out?